python/cudf/cudf/core/dataframe.py

```diff
@@ -547,7 +548,10 @@ def deserialize(cls, header, frames):
         column_names = pickle.loads(header["column_names"])
         columns = column.deserialize_columns(header["columns"], column_frames)
 
-        return cls(dict(zip(column_names, columns)), index=index)
+        return cls._from_data(
+            {name: col for name, col in zip(column_names, columns)},
```

Comment:
The `{}` syntax is generally faster than `dict` for construction because `dict` incurs function-call overhead, but a comprehension incurs unnecessary additional overhead here:

```
In [1]: x = list(range(10000))

In [2]: %timeit dict(zip(x, x))
490 µs ± 12.9 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [3]: %timeit {k: v for k, v in zip(x, x)}
963 µs ± 792 ns per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

```suggestion
            dict(zip(column_names, columns)),
```